crates/red_knot_python_semantic/src/types/property_tests.rs

```diff
@@ -305,6 +310,14 @@ mod stable {
         never_subtype_of_every_fully_static_type, db,
         forall types t. t.is_fully_static(db) => Type::Never.is_subtype_of(db, t)
     );
+
+    // For fully static types, they should be subtype of their union.
+    type_property_test!(
+        fully_static_types_are_subtype_of_their_union, db,
+        forall types s, t.
+            s.is_fully_static(db) && t.is_fully_static(db)
+            => s.is_subtype_of(db, union(db, s, t)) && t.is_subtype_of(db, union(db, s, t))
+    );
```

Comment:
Is it worth adding another test as well for assignability with non-fully-static types?

```suggestion
    // For fully static types, they should be subtype of their union.
    type_property_test!(
        fully_static_types_are_subtype_of_their_union, db,
        forall types s, t.
            s.is_fully_static(db) && t.is_fully_static(db)
            => s.is_subtype_of(db, union(db, s, t)) && t.is_subtype_of(db, union(db, s, t))
    );
    
    // And for *any* two types, even non-fully-static types,
    // each of the pair they should be assignable to the union of the two.
    type_property_test!(
        all_type_pairs_are_assignable_to_their_union, db,
        forall types s, t. s.is_assignable_to(db, union(db, s, t)) && t.is_assignable_to(db, union(db, s, t))
    );
```

Reply:
That makes sense! I’ve added it. However, `all_type_pairs_are_assignable_to_their_union` was failing with high probability, so I moved it to the flaky module. It seems that `is_assignable_to` is not fully implemented for non-fully-static types.